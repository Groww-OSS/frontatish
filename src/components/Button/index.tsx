// base and lib imports
import React, { useState } from 'react';
import { Text, View, LayoutChangeEvent } from 'react-native';
import Ripple from 'react-native-material-ripple';

// utils and helpers
import { StyleType } from '../../common/types';
import { ButtonProps } from './types';
import { getBtnStyles, getLabelStyles } from './helper';

// styles and themes
import { useColors } from '../../themes';

import Progress from '../../animated/Progress';

const Button = (props: ButtonProps) => {
  const { customStyles, disabled, label, type, onPress, loading } = props;
  const [width, setWidth] = useState<number>(0);
  // getting the suitable color based on the theme
  // activated inside the app
  const Colors = useColors();
  // getting base button styles defined
  // by our design guidelines, component
  // also takes custom styles so that can also
  // be applied
  const baseBtnStyles: StyleType = getBtnStyles(type, Colors, disabled);
  const mainBtnStyles: StyleType = {
    ...baseBtnStyles,
    ...customStyles,
  };
  const baseLabelStyles: StyleType = getLabelStyles(type, Colors, disabled);
  const measureLayout = (e: LayoutChangeEvent) => {
    setWidth(e.nativeEvent.layout.width);
  };

  const renderProgressBar = () => {
    const barColor = type === 'primary' ? Colors.white : Colors.primary;
    return (
      <View
        style={{
          position: 'absolute',
          top: 0,
          borderRadius: 5,
        }}
      >
        <Progress width={width} barColor={barColor} />
      </View>
    );
  };

  return (
    <Ripple
      onPress={onPress}
      disabled={disabled || loading}
      style={mainBtnStyles}
      onLayout={measureLayout}
      accessibilityRole="button"
      accessibilityState={{ disabled: disabled || loading }}
      accessibilityLabel={label}
    >
      {loading && renderProgressBar()}
      <Text style={baseLabelStyles}>{label}</Text>
    </Ripple>
  );
};

// const styles = StyleSheet.create({
//   // define styles
//   // btnContainer: {
//   //   height: 50,
//   //   flex: 1,
//   //   marginVertical: 20,
//   //   borderRadius: 4,
//   //   alignItems: 'center',
//   //   justifyContent: 'center',
//   // },
//   // btnText: {
//   //   fontFamily: Fonts.type.gotham_medium,
//   //   fontSize: Fonts.size.regular_16,
//   //   fontWeight: 'bold',
//   // }
// });

Button.defaultProps = {
  children: null,
  disabled: false,
  label: '',
  loading: false,
  type: 'primary',
  onPress: null,
  customStyles: {},
};

export default Button;
