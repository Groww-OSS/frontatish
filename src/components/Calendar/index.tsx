import React, { useState } from 'react';
import { View, Text, StyleSheet, ScrollView, SafeAreaView } from 'react-native';
// eslint-disable-next-line import/no-unresolved
import IonIcon from 'react-native-vector-icons/Ionicons';
import Ripple from 'react-native-material-ripple';
import Modalbox from 'react-native-modalbox';
import { Button } from 'frontatish';
import { useColors } from '../../themes';
import { CalendarProps } from './types';
import { Fonts } from '../../styles';

const styles = StyleSheet.create({
  calendarContainer: {
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,
    elevation: 3,
  },
  questionText: {
    fontSize: 18,
    fontWeight: '500',
    fontStyle: 'normal',
    lineHeight: 24,
    letterSpacing: -0.17,
    textAlign: 'center',
    marginTop: 20,
  },
  yearSelectedStyle: {
    height: 60,
    width: 60,
    justifyContent: 'center',
    alignSelf: 'center',
    alignItems: 'center',
    fontSize: 21,
    fontWeight: '400',
    fontStyle: 'normal',
    letterSpacing: 0,
    textAlign: 'center',
    paddingVertical: 10,
    // marginBottom: 4,
    marginRight: 10,
    color: '#44475B',
  },
  unselectedStyle: {
    height: 40,
    width: 40,
    fontFamily: Fonts.type.gotham_medium,
    fontSize: 13,
    fontWeight: '500',
    fontStyle: 'normal',
    letterSpacing: 0,
    textAlign: 'center',
    color: 'rgba(0, 0, 0, 0.5)',
    paddingVertical: 10,
    marginBottom: 4,
    marginRight: 10,
    borderRadius: 25,
    borderStyle: 'solid',
    borderWidth: 1,
    borderColor: '#FFFFFF',
  },
  selectedStyle: {
    height: 40,
    width: 40,
    borderRadius: 25,
    borderStyle: 'solid',
    backgroundColor: '#00D09C',
    borderWidth: 1,
    borderColor: '#dfe1e4',
    fontFamily: Fonts.type.gotham_medium,
    fontSize: 13,
    fontWeight: '500',
    fontStyle: 'normal',
    letterSpacing: 0,
    textAlign: 'center',
    color: '#FFFFFF',
    paddingVertical: 10,
    marginBottom: 4,
    marginRight: 10,
    overflow: 'hidden',
  },
});

const months = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

const nDays = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];

const Calendar = (props: CalendarProps) => {
  const [activeDate, setActiveDate] = useState(new Date());
  const [selectedYear, setSelectedYear] = useState(new Date().getFullYear());
  const [selectedMonth, setSelectedMonth] = useState(new Date().getMonth());
  // getting the suitable color based on the theme
  // activated inside the app
  const Colors = useColors();
  const {
    setDate,
    title,
    children,
    isDayRequired,
    yearsArray,
    isOpen,
    onClosed,
    onConfirmClick,
    picker_key,
  } = props;
  const changeMonth = (delta: number) => {
    const newTimeInMS = activeDate.setMonth(activeDate.getMonth() + delta);
    const updatedDate = new Date(newTimeInMS);
    setActiveDate(updatedDate);
  };
  const changeDay = (dateOfMonth: number) => {
    const newTimeInMS = activeDate.setMonth(activeDate.getMonth(), dateOfMonth);
    const updatedDate = new Date(newTimeInMS);
    setActiveDate(updatedDate);
  };
  setDate(activeDate);

  const generateMatrix = () => {
    const matrix = [];
    // Create header
    matrix[0] = ['1', '2', '3', '4', '5', '6', '7'];
    const year = activeDate.getFullYear();
    const month = activeDate.getMonth();

    // const firstDay = new Date(year, month, 1).getDay();
    let maxDays = nDays[month];
    if (month === 1) {
      // February
      if ((year % 4 === 0 && year % 100 !== 0) || year % 400 === 0) {
        maxDays += 1;
      }
    }
    let counter = 8;
    for (let row = 1; row < 5; row += 1) {
      matrix[row] = [];
      for (let col = 0; col < 7; col += 1) {
        if (counter <= maxDays) {
          matrix[row][col] = counter.toString();
          counter += 1;
        } else {
          matrix[row][col] = '';
        }
      }
    }
    return matrix;
  };
  const matrix = generateMatrix();
  const renderEachRow = (daysArray: string[]) => {
    // console.log('dataArray', daysArray);
    return (
      <View
        style={{
          flexDirection: 'row',
          justifyContent: 'space-evenly',
          alignItems: 'center',
        }}
      >
        {daysArray.map((item) => {
          return item === '' ? null : (
            <Ripple
              onPress={() => changeDay(parseInt(item, 10))}
              rippleContainerBorderRadius={18}
            >
              <View
                style={{
                  justifyContent: 'center',
                  borderRadius: 18,
                  height: 36,
                  width: 36,
                  backgroundColor:
                    activeDate.getDate().toString() === item
                      ? Colors.primary_attr_90
                      : Colors.white,
                }}
              >
                <Text
                  style={{
                    color: Colors.font_2,
                    textAlign: 'center',
                  }}
                >
                  {item}
                </Text>
              </View>
            </Ripple>
          );
        })}
      </View>
    );
  };
  return (
    <View>
      {isDayRequired ? (
        <View
          style={[
            styles.calendarContainer,
            { backgroundColor: Colors.white, shadowColor: Colors.font_1 },
          ]}
        >
          <View style={{ marginVertical: 16 }}>
            <Text
              style={{
                textAlign: 'center',
                color: Colors.font_1,
                fontWeight: 'bold',
              }}
            >
              {title}
            </Text>
          </View>
          <View
            style={{
              height: 40,
              backgroundColor: Colors.font_6,
              justifyContent: 'center',
            }}
          >
            <View
              style={{
                flexDirection: 'row',
                marginHorizontal: 30,
                alignItems: 'center',
              }}
            >
              <Ripple
                style={{
                  height: 32,
                  width: 32,
                  alignItems: 'center',
                  justifyContent: 'center',
                }}
                rippleContainerBorderRadius={16}
                onPress={() => changeMonth(-1)}
              >
                <IonIcon
                  name="ios-arrow-back"
                  size={16}
                  color={Colors.primary}
                  // style={{ flex: 1, textAlign: 'center' }}
                />
              </Ripple>
              <Text
                style={{
                  color: Colors.primary,
                  flex: 2,
                  textAlign: 'center',
                  fontWeight: '500',
                  fontSize: 16,
                }}
              >
                {months[activeDate.getMonth()]}
              </Text>
              <Ripple
                style={{
                  height: 32,
                  width: 32,
                  alignItems: 'center',
                  justifyContent: 'center',
                }}
                rippleContainerBorderRadius={16}
                onPress={() => changeMonth(1)}
              >
                <IonIcon
                  name="ios-arrow-forward"
                  size={16}
                  color={Colors.primary}
                  // style={{ flex: 1, textAlign: 'center' }}
                />
              </Ripple>
            </View>
          </View>
          <View style={{ alignItems: 'center' }}>
            <View style={{ alignItems: 'flex-start', marginTop: 16 }}>
              {matrix.map((item) => renderEachRow(item))}
            </View>
          </View>
          {children}
        </View>
      ) : (
        <Modalbox
          style={{
            width: 302,
            height: null,
            borderRadius: 10,
            paddingHorizontal: 20,
            maxHeight: 320,
          }}
          isOpen={isOpen}
          // isDisabled={isDisabled}
          backdropPressToClose
          swipeToClose={false}
          position="center"
          entry="bottom"
          backdrop
          backdropOpacity={0.8}
          backdropColor="black"
          backdropContent={null}
          animationDuration={400}
          onClosed={onClosed}
          backButtonClose
          startOpen={false}
          coverScreen
          // keyboardTopOffset={22}
        >
          <Text style={styles.questionText}>Select a year & month</Text>
          <ScrollView
            pinchGestureEnabled={false}
            horizontal
            style={{
              marginHorizontal: 30,
            }}
            contentContainerStyle={{
              marginTop: 15,
              flexGrow: 1,
              alignItems: 'center',
              justifyContent: 'center',
              flexWrap: 'wrap',
            }}
            // style={{
            //   backgroundColor: '#fff',
            //   borderTopLeftRadius: 100,
            //   borderTopRightRadius: 100,
            //   //flexDirection: 'column',
            //   paddingHorizontal: 20,
            // }}
            // contentContainerStyle={{
            //   flexGrow: 1,
            //   paddingHorizontal: 20,
            //   paddingVertical: 10,
            //   backgroundColor: '#ff1',
            // }}
            showsHorizontalScrollIndicator={false}
          >
            {yearsArray.map((item, index) => (
              <Text
                key={index}
                style={
                  selectedYear === item
                    ? [styles.yearSelectedStyle, { marginBottom: 0, flex: 1 }]
                    : [
                        styles.unselectedStyle,
                        {
                          marginBottom: 0,
                          flex: 1,
                          width: 60,
                        },
                      ]
                }
                onPress={() => setSelectedYear(item)}
              >
                {item}
              </Text>
            ))}
          </ScrollView>
          <View
            style={{
              flexDirection: 'row',
              flexWrap: 'wrap',
              justifyContent: 'center',
              paddingHorizontal: 25,
            }}
          >
            {months.map((item, index) => {
              return (
                <Text
                  key={index}
                  onPress={() => setSelectedMonth(index)}
                  style={
                    months[selectedMonth] === item
                      ? [styles.selectedStyle]
                      : [styles.unselectedStyle]
                  }
                >
                  {item.substr(0, 3)}
                </Text>
              );
            })}
          </View>
          <Button
            label="Done"
            customStyles={{ margin: 12 }}
            onPress={() =>
              onConfirmClick({
                key: picker_key,
                value: { year: selectedYear, month: selectedMonth },
              })
            }
          />
        </Modalbox>
      )}
    </View>
  );
};

Calendar.defaultProps = {
  title: 'Select Date:',
  isDayRequired: true,
  yearsArray: [2016, 2017, 2018, 2019, 2020, 2021],
  isOpen: false,
  onClosed: () => {},
  onConfirmClick: () => {},
  picker_key: {},
};

export default Calendar;
