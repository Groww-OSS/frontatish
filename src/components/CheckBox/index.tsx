// base and lib imports
import React from 'react';
// eslint-disable-next-line import/no-unresolved
import Icon from 'react-native-vector-icons/MaterialIcons';
import { StyleSheet, View } from 'react-native';

// types
import { CheckBoxProps } from './types';

// utils and helpers
import { getCheckDmnsn } from '../../common/utils';

// common components
import { Scale } from '../../animated';

// styles and themes
import { useColors } from '../../themes';

// This API is broken is terms of accessible because for now checkbox only shows a box element which in the end will be wrapped inside pressable,
// resulting into nested element which won't be covered by touches hence resulting into not accessible
// Refer CheckBox, the api should be similar where labels are passed as props
const CheckBox = (props: CheckBoxProps) => {
  const { checked, containerStyle, disabled, size, checkColor } = props;
  const [height, width, iconSize] = getCheckDmnsn(size!);
  const Colors = useColors();
  const getRequiredColor = () => {
    if (disabled) {
      if (!checked) {
        return [Colors.white, Colors.font_3, Colors.white];
      }
      return [Colors.font_3, Colors.font_3, Colors.white];
    }
    if (!disabled && checked) {
      return [Colors.primary, Colors.transparent, Colors.white];
    }
    return [Colors.white, Colors.primary];
  };

  const [backgroundColor, borderColor, iconColor] = getRequiredColor();
  // if check color exists use that only,else fallback to iconColor
  // returned from our main color fetch method.
  const finalIconColor = checkColor || iconColor;
  const mainContainerStyle = {
    ...styles.boxContainer,
    backgroundColor,
    borderColor,
    // giving preference to custom styles
    ...containerStyle,
    height,
    width,
  };
  return (
    <View
      style={mainContainerStyle}
      accessibilityRole="checkbox"
      accessibilityState={{ checked, disabled }}
    >
      {/* <Fade visible={checked!} containerStyle={{ flex: 1 }} duration={300}>
          <Icon name="check" size={iconSize} color={iconColor} />
        </Fade> */}

      <Scale scale={checked!!}>
        <Icon name="check" size={iconSize} color={finalIconColor} />
      </Scale>
    </View>
  );
};

const styles = StyleSheet.create({
  boxContainer: {
    borderWidth: 2,
    borderRadius: 4,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
export default CheckBox;
