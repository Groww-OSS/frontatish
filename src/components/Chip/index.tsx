import React from 'react';
import { StyleSheet, Text, View, TouchableWithoutFeedback } from 'react-native';
// eslint-disable-next-line import/no-unresolved
import IonIcon from 'react-native-vector-icons/Ionicons';

// eslint-disable-next-line import/no-unresolved
import { ChipProps } from './types';

const Chip = (props: ChipProps) => {
  const {
    label,
    onPress,
    disabled,
    iconName,
    iconStyle,
    onClose,
    isSelected,
    labelStyle,
    containerStyle,
    selectedContainerStyle,
  } = props;

  return (
    <View
      style={[
        styles.container,
        containerStyle,
        isSelected ? [styles.selectedContainer, selectedContainerStyle] : {},
      ]}
    >
      <TouchableWithoutFeedback disabled={disabled} onPress={onPress}>
        <View style={styles.iconLabelContainer}>
          {iconName ? (
            <IonIcon name={iconName} style={[styles.icon, iconStyle]} />
          ) : null}

          <Text style={[styles.label, labelStyle]}>{label}</Text>
        </View>
      </TouchableWithoutFeedback>

      {onClose ? (
        <TouchableWithoutFeedback onPress={onClose}>
          <IonIcon name="ios-close-circle" style={styles.closeIcon} size={15} />
        </TouchableWithoutFeedback>
      ) : null}
    </View>
  );
};

Chip.defaultProps = {
  label: 'Chip',
  disabled: false,
  iconName: null,
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 30,
    paddingHorizontal: 8,
    paddingVertical: 3,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: '#ECEDEF',
  },
  selectedContainer: {
    backgroundColor: '#00D09C',
  },
  label: {
    fontFamily: 'Roboto',
    fontWeight: '500',
    fontSize: 12,
    letterSpacing: 0.01,
    lineHeight: 19,
  },
  iconLabelContainer: { flexDirection: 'row', alignItems: 'center' },
  icon: {
    paddingRight: 4,
  },
  closeIcon: { paddingLeft: 4, paddingVertical: 3 },
});

export default Chip;
