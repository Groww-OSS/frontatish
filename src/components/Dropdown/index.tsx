import React, { useRef, useState } from 'react';
import {
  View,
  Text,
  StyleSheet,
  Animated,
  TouchableNativeFeedback,
} from 'react-native';
// eslint-disable-next-line import/no-unresolved
import Icon from 'react-native-vector-icons/MaterialIcons';
import { useColors } from '../../themes';
import { ColorType } from '../../common/types';
import { scaleDimension } from '../../common/utils';
import { DropdownProps } from './types';
import DropList from './DropList';
import Line from './Line';

const Dropdown = (props: DropdownProps) => {
  const [open, setOpen] = useState(false);
  const { items, active, onChange, itemStyle, activeItemStyle, iconColor } =
    props;
  const animateRotate = useRef(new Animated.Value(0)).current;
  const Colors = useColors();
  const styles = getStyles(Colors);

  const handleList = () => {
    // axios.get();
    Animated.timing(animateRotate, {
      toValue: open ? 0 : 1,
      duration: 300,
      useNativeDriver: true,
    }).start();
    setOpen(!open);
  };
  const onItemClick = (index: number) => {
    onChange(index);
    handleList();
  };
  const itemLabel = items[active]?.label ?? items[active];
  return (
    <View style={styles.container}>
      <TouchableNativeFeedback
        onPress={handleList}
        accessibilityTraits="button"
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={{ expanded: open }}
        accessibilityLabel={itemLabel}
      >
        <View style={styles.activeItem}>
          <Text style={{ color: Colors.font_2, ...activeItemStyle }}>
            {itemLabel}
          </Text>
          <View style={{ flex: 1, alignItems: 'flex-end' }}>
            <Animated.View
              style={[
                {
                  transform: [
                    {
                      rotate: animateRotate.interpolate({
                        inputRange: [0, 1],
                        outputRange: ['0deg', '180deg'],
                      }),
                    },
                  ],
                },
              ]}
            >
              <Icon
                name="keyboard-arrow-down"
                size={scaleDimension(20)}
                color={iconColor ?? Colors.primary}
              />
            </Animated.View>
          </View>
        </View>
      </TouchableNativeFeedback>
      {!open && <Line />}
      {open && (
        <DropList
          items={items}
          active={active}
          onChange={onItemClick}
          itemStyle={itemStyle}
        />
      )}
    </View>
  );
};

const getStyles = (Colors: ColorType) => {
  return StyleSheet.create({
    container: { margin: 20 },
    activeItem: {
      flexDirection: 'row',
      alignItems: 'center',
      paddingVertical: 10,
    },
    caret: {
      width: 0,
      height: 0,
      borderLeftWidth: 6,
      borderRightWidth: 6,
      borderBottomWidth: 12,
      borderStyle: 'solid',
      backgroundColor: 'transparent',
      borderLeftColor: 'transparent',
      borderRightColor: 'transparent',
      borderBottomColor: Colors.primary,
    },
  });
};

export default Dropdown;
