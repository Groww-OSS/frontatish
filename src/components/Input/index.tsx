import React from 'react';
import {
  TextInput,
  TextInputProps,
  View,
  StyleSheet,
  Text,
  TouchableWithoutFeedback,
} from 'react-native';
// eslint-disable-next-line import/no-unresolved

import { withColors } from '../../themes';
import { StyleType, ColorType } from '../../common/types';

let styles: any;
interface InputProps extends TextInputProps {
  label?: string;
  error?: string;
  disabled?: string;
  labelStyle?: StyleType;
  inputTextStyle?: StyleType;
  disabledTextStyle?: StyleType;
  containerStyle?: StyleType;
  bottomBorderStyle?: StyleType;
  Colors?: any;
}

interface InputState {
  active: boolean;
}

class Input extends React.PureComponent<InputProps, InputState> {
  textInput: React.RefObject<TextInput>;

  constructor(props: InputProps) {
    super(props);
    this.state = { active: false };
    this.textInput = React.createRef();
  }

  setActive = (active: boolean) => {
    this.setState({ active });
  };

  setFocus = () => {
    this.textInput?.current?.focus();
  };

  getBottomBorderStyles = () => {
    const { bottomBorderStyle, Colors, error } = this.props;
    const { active } = this.state;
    let mergedStyle;
    if (error) {
      mergedStyle = [
        bottomBorderStyle,
        {
          height: 2,
          // marginTop: 4,
          backgroundColor: Colors.semantic_red,
        },
      ];
    } else if (active) {
      mergedStyle = [
        bottomBorderStyle,
        {
          height: 2,
          // marginTop: 4,
          backgroundColor: Colors.primary,
        },
      ];
    } else {
      mergedStyle = [
        bottomBorderStyle,
        {
          height: 1,
          // marginTop: 4,
          backgroundColor: Colors.font_5,
        },
      ];
    }
    return mergedStyle;
  };

  render() {
    const {
      label,
      error,
      containerStyle,
      labelStyle,
      Colors,
      disabled,
      disabledTextStyle,
      inputTextStyle,
    } = this.props;
    styles = getStyles(Colors);
    const bottomBorderStyles = this.getBottomBorderStyles();
    const textInputStyle = disabled
      ? { ...styles.disabledTextStyle, ...disabledTextStyle }
      : { ...styles.inputTextStyle, ...inputTextStyle };
    return (
      <TouchableWithoutFeedback onPress={this.setFocus}>
        <View style={[styles.containerStyle, containerStyle]}>
          <Text style={[styles.lableStyle, labelStyle]}>{label}</Text>
          <TextInput
            {...this.props}
            style={textInputStyle}
            // eslint-disable-next-line no-return-assign
            ref={this.textInput}
            editable={!disabled}
            onFocus={() => this.setActive(true)}
            onBlur={() => this.setActive(false)}
          />
          <View style={bottomBorderStyles} />
          {error && <Text style={styles.errorStyle}>{error}</Text>}
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

const getStyles = (Colors: ColorType) => {
  return StyleSheet.create({
    containerStyle: { padding: 20 },
    lableStyle: { color: Colors.font_2, marginBottom: 8, fontWeight: '500' },
    errorStyle: { color: Colors.semantic_red, marginTop: 4, fontWeight: '500' },
    inputTextStyle: { color: Colors.font_1 },
    disabledTextStyle: { color: Colors.font_3 },
  });
};

export default withColors(Input);
