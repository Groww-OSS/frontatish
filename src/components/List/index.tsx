import React, { useMemo } from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';
import Ripple from 'react-native-material-ripple';

// eslint-disable-next-line import/no-unresolved
import IonIcon from 'react-native-vector-icons/Ionicons';
// eslint-disable-next-line import/no-unresolved
import { Images, useColors, Chip } from 'frontatish';
import { ListProps } from './types';

const List = (props: ListProps) => {
  const Colors = useColors();
  const {
    leftLabel,
    leftSubtext,
    rightLabel,
    rightSubtext,
    leftChipLabel,
    rightChipLabel,
    sparkLine,
    disabled,
    onPress,
    labelStyle,
    subtextStyle,
    containerStyle,
  } = props;

  const getChip = (chipLabel: string | undefined) => {
    return (
      <>
        {chipLabel ? (
          <Chip label={chipLabel} containerStyle={styles.chip} />
        ) : null}
      </>
    );
  };
  const getLabelSubtextChip = (
    label: string,
    subtext: string | undefined,
    chipLabel: string | undefined,
  ) => {
    return (
      <>
        <Text style={[styles.label, { color: Colors.font_1 }, labelStyle]}>
          {label}
        </Text>
        {subtext ? (
          <Text
            style={[styles.subtext, { color: Colors.font_1 }, subtextStyle]}
          >
            {subtext}
          </Text>
        ) : null}
        {useMemo(() => getChip(chipLabel), [chipLabel])}
      </>
    );
  };

  const getLeftLabelSubtextView = (
    label: string,
    subtext: string | undefined,
    chipLabel: string | undefined,
  ) => {
    return (
      <View style={styles.leftView}>
        {getLabelSubtextChip(label, subtext, chipLabel)}
      </View>
    );
  };
  const getRightLabelSubtextView = (
    label: string,
    subtext: string | undefined,
    chipLabel: string | undefined,
  ) => {
    return (
      <View style={styles.rightView}>
        {getLabelSubtextChip(label, subtext, chipLabel)}
      </View>
    );
  };

  return (
    <View>
      <Ripple
        onPress={onPress}
        disabled={disabled}
        style={[
          styles.container,
          { backgroundColor: Colors.white },
          containerStyle,
        ]}
      >
        {getLeftLabelSubtextView(leftLabel, leftSubtext, leftChipLabel)}
        {sparkLine ? (
          <View>
            <Image
              source={Images.sparkLine}
              style={styles.sparkLine}
              resizeMethod="auto"
            />
          </View>
        ) : null}
        {rightLabel
          ? getRightLabelSubtextView(rightLabel, rightSubtext, rightChipLabel)
          : null}
        {!disabled ? (
          <IonIcon
            name="ios-arrow-forward"
            size={10}
            style={styles.arrowIcon}
          />
        ) : null}
      </Ripple>
    </View>
  );
};

List.defaultProps = {
  sparkLine: false,
  disabled: false,
};

const styles = StyleSheet.create({
  label: { fontSize: 16 },
  subtext: { fontSize: 12, marginTop: 6 },
  chip: { marginTop: 6 },
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomWidth: 1,
    paddingVertical: 20,
    paddingHorizontal: 12,
  },
  leftView: {
    flex: 4,
    alignItems: 'flex-start',
  },
  rightView: {
    flex: 3,
    alignItems: 'flex-end',
  },
  sparkLine: { height: 54, width: 92 },
  arrowIcon: { paddingRight: 8, paddingLeft: 14 },
});

export default List;
