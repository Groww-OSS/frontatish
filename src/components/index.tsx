// eslint-disable-next-line import/no-unresolved
import Ripple from 'react-native-material-ripple';
import Button from './Button';
import CheckBox from './CheckBox';
import Calendar from './Calendar';
import Chip from './Chip';
import Dropdown from './Dropdown';
import Input from './Input';
import List from './List';
import NumPad from './NumPad';
import RadioButton from './RadioButton';
import Table from './Table';
import EmptyState from './EmptyState';
import EmptyView from './EmptyView';
import Switch from './Switch';
import Searchbar from './Searchbar';
import Tag from './Tag';

export {
  Button,
  CheckBox,
  Calendar,
  Chip,
  EmptyState,
  EmptyView,
  Dropdown,
  List,
  NumPad,
  RadioButton,
  Ripple,
  Switch,
  Searchbar,
  Table,
  Tag,
  Input,
};
