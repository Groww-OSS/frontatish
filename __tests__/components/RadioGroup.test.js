import 'react-native';
import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { RadioGroup, RadioButton } from '../../src/components';

it('Renders Correctly', () => {
  const tree = toJson(
    shallow(
      <RadioGroup>
        <RadioButton value="value1" />
        <RadioButton value="value2" />
      </RadioGroup>,
    ),
  );
  expect(tree).toMatchSnapshot();
});

it('RadioButton onPress Triggers', () => {
  const onPressFn = jest.fn();
  const component = shallow(<RadioButton value="test" onPress={onPressFn} />);
  component.simulate('press');
  expect(onPressFn).toBeCalledWith('test');
});

it('RadioGroup state updates correctly', () => {
  const component = shallow(
    <RadioGroup>
      <RadioButton value="value1" />
      <RadioButton value="value2" />
      <RadioButton value="value3" />
    </RadioGroup>,
  );

  component.find(RadioButton).at(0).prop('onPress')('value1'); // press RadioButton 1

  const radioButtons = component.find(RadioButton);
  radioButtons.forEach((radioButton) => {
    if (radioButton.prop('value') === 'value1')
      expect(radioButton.prop('selected')).toBeTruthy();
    else expect(radioButton.prop('selected')).toBeFalsy();
  });
});
