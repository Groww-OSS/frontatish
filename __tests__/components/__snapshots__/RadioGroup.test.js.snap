// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<View>
  <RadioButton
    key="0"
    onPress={[Function]}
    selected={false}
    size="sm"
    value="value1"
  />
  <RadioButton
    key="1"
    onPress={[Function]}
    selected={false}
    size="sm"
    value="value2"
  />
</View>
`;
